### ISSUE TITLE
vault config file should support settings to implement local default values for -key-shares and -key-threshold during init

### ISSUE THREAD

**Issue by careyblack** [hashicorp/vault#722]
I think it would be useful to include the -key-shares and -key-threshold values for a vault server in the servers config file.

1) This way the local file can be the "default values" and the "vault init" command can be bare of parameters and you still get what you want for "-key-shares and -key-threshold" values.

2) It is a way for config management to push/control a very fundamental aspect of standing up a new vault.

Maybe you could do something as simple as:
"
secret_shares=integer
secret_threshold=integer
"

In similar fashion to the "disable_mlock=boolean" config value.

These settings should also be honored during rekey operations too.

**Comment by careyblack** [comment]
Oh.. and the values should also be defaults during "rekey" processing too.

**Comment by jefferai** [comment]
This could be implemented in the future. However, you should be able to handle these parameters successfully with configuration management now. Are you having issues with it?

**Comment by careyblack** [comment]
Well... given that my current configuration management tool is the oldest one in the tool box, yes I appear to have a problem. Maybe you have heard of it? It is called "me". :) And I like having config files that fully describe the desired state for a tool. As the project is, the config file only partially defines the state (focused on run time state and appears to ignore initialization values) and requires more intelligence to be build into the configuration management tool that is establishing and maintaining vault.  That external dependency can be a barrier to adoption due to poor support for those external dependencies. Not to mention that it is harder to communicate "best practices" for command line options (and why you use "X" instead of "y")  in user docs verses commented configuration files.  ( IMHO )
 

I think that having a configuration file that defines (at least the desired) defaults just makes sense. 

Only when the file is missing should hard coded defaults in the tools/code possibly be used. (Or an error should be raised if it is determined that the user MUST supply a value. Sometimes it is necessary for the user to pick their poison so that they only have themselves to blame.) 

In short, ( I know to late) any inability to control "magic strings" in the code from a config file seems like a "miss" in the feature set to me.
